pkg/controller/controller.go

```diff
@@ -277,55 +277,75 @@ func (s *Service) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest
+					}
 
-				group, err := arr.Client.CreateVolumeGroup(ctx, &gopowerstore.VolumeGroupCreate{
-					Name:               vgName,
-					ProtectionPolicyID: pp,
-				})
-				if err != nil {
-					return nil, status.Errorf(codes.Internal, "can't create volume group: %s", err.Error())
-				}
+					group, err := arr.Client.CreateVolumeGroup(ctx, &gopowerstore.VolumeGroupCreate{
+						Name:               rgName,
+						ProtectionPolicyID: pp,
+					})
+					if err != nil {
+						return nil, status.Errorf(codes.Internal, "can't create volume group: %s", err.Error())
+					}
 
-				vg, err = arr.Client.GetVolumeGroup(ctx, group.ID)
-				if err != nil {
-					return nil, status.Errorf(codes.Internal, "can't query volume group by id %s : %s", group.ID, err.Error())
-				}
+					vg, err = arr.Client.GetVolumeGroup(ctx, group.ID)
+					if err != nil {
+						return nil, status.Errorf(codes.Internal, "can't query volume group by id %s : %s", group.ID, err.Error())
+					}
 
-			} else {
-				return nil, status.Errorf(codes.Internal, "can't query volume group by name %s : %s", vgName, err.Error())
-			}
-		} else {
-			// group exists, check that protection policy applied
-			if vg.ProtectionPolicyID == "" {
-				pp, err := EnsureProtectionPolicyExists(ctx, arr, vgName, remoteSystemName, rpoEnum)
-				if err != nil {
-					return nil, status.Errorf(codes.Internal, "can't ensure protection policy exists %s", err.Error())
+				} else {
+					return nil, status.Errorf(codes.Internal, "can't query volume group by name %s : %s", rgName, err.Error())
 				}
-				policyUpdate := gopowerstore.VolumeGroupChangePolicy{ProtectionPolicyID: pp}
-				_, err = arr.Client.UpdateVolumeGroupProtectionPolicy(ctx, vg.ID, &policyUpdate)
-				if err != nil {
-					return nil, status.Errorf(codes.Internal, "can't update volume group policy %s", err.Error())
+			} else {
+				// group exists, check that protection policy applied
+				if vg.ProtectionPolicyID == "" {
+					pp, err := EnsureProtectionPolicyExists(ctx, arr, rgName, remoteSystemName, rpoEnum)
+					if err != nil {
+						return nil, status.Errorf(codes.Internal, "can't ensure protection policy exists %s", err.Error())
+					}
+					policyUpdate := gopowerstore.VolumeGroupChangePolicy{ProtectionPolicyID: pp}
+					_, err = arr.Client.UpdateVolumeGroupProtectionPolicy(ctx, vg.ID, &policyUpdate)
+					if err != nil {
+						return nil, status.Errorf(codes.Internal, "can't update volume group policy %s", err.Error())
+					}
 				}
 			}
-		}
 
-		if c, ok := creator.(*SCSICreator); ok {
-			c.vg = &vg
+			if c, ok := creator.(*SCSICreator); ok {
+				c.vg = &vg // Only add VG to the creator in the case of non-NFS volumes.
+			}
+		} else { // NFS volumes use protected NAS servers instead of protected VGs for replication.
+
+			// Create the protection policy if it does not already exist
+			pp, err := EnsureProtectionPolicyExists(ctx, arr, rgName, remoteSystemName, rpoEnum)
```

Comment:
When you work on the next story, please check if you can refactor calling 'EnsureProtectionPolicyExists' to avoid code duplication.